cpp/src/table/table.cpp

```diff
@@ -79,5 +79,25 @@ std::vector<std::unique_ptr<column>> table::release() {
   return std::move(_columns);
 }
 
+// Returns a table_view with set of specified columns
+table_view table::select(std::vector<gdf_size_type> const& column_indices) const {
+    CUDF_EXPECTS(column_indices.size() <= _columns.size(), "Requested too many columns.");
+
+    std::vector<column_view> columns;
+    for (auto index : column_indices) {
+      columns.push_back(_columns.at(index)->view());
```

Comment:
Update doc to indicate that this throws if `column_indices` has any index outside of `[0, num_columns())`.

Reply:
Updated [here](https://github.com/rapidsai/cudf/pull/3065/files#diff-6e2840973c2915fb2c562344d98e0886R120).